product_state/models/product_template.py

```diff
@@ -55,10 +55,10 @@ def _set_product_state_id(self, record):
         ProductState = record.env["product.state"]
         product_state = ProductState.search([("code", "=", record.state)], limit=1)
         if record.state and not product_state:
-            msg = _("The product state code %s could not be found.")
+            msg = self.env._("The product state code %s could not be found.")
```

Comment:
Use the translation function arguments instead.

Reply:
this is the new way to translate :wink: 
https://www.odoo.com/documentation/18.0/developer/howtos/translations.html#explicit-exports

Reply:
Nope. I didn't speak about that.

```suggestion
            msg = self.env._("The product state code %(product_state)s could not be found.", product_state=record.state)
```